lua/neogit/status.lua

```diff
@@ -316,25 +317,41 @@ function refresh(force)
     return wait
   end
 
-  a.dispatch(function ()
+  run(future(function ()
     refreshing = true
 
-    if a.wait(cli.git_root()) ~= '' then
-      a.wait(repo:update_status())
-      a.wait_all({
-        repo:update_branch_information(),
-        repo:update_stashes(),
-        repo:update_unpulled(),
-        repo:update_unmerged(),
-        repo:load_diffs()
-      })
-      a.wait_for_textlock()
+    local time = os.clock()
+    if await(cli.git_root()) ~= '' then
+      await(repo:update_status())
+      print('reload 1', os.clock() - time)
+      await(scheduler())
       refresh_status()
+
+      await_all({
+        future(function ()
+          await_all({
+            repo:update_branch_information(),
+            repo:update_stashes(),
+            repo:update_unpulled(),
+            repo:update_unmerged(),
+          })
+          print('reload info', os.clock() - time)
+          await(scheduler())
+          refresh_status()
+        end),
+        future(function ()
+          await(repo:load_diffs())
+          print('reload diff', os.clock() - time)
+          await(scheduler())
+          refresh_status()
+        end)
+      })
       vim.cmd [[do <nomodeline> User NeogitStatusRefreshed]]
     end
+    print('reload', os.clock() - time)
```

Comment:
Hey @oberblastmeister, quick question: I noticed an ever so slight delay in the initial buffer draw and tried to improve the time to the initial redraw by staggering the loading, but that didn't seem to have any effect. I assume that neovim doesn't redraw the buffer until the lua coroutine finished unwinding?
Do you have any insight into that or know how to force neovim to update the buffer? If not I think I can deal with that some other way, just thought I'd ask.

Reply:
@RianFuro I also noticed a slight delay when opening the status, but sometimes it is instant for me (Not sure why).

Reply:
The answer to one of my questions is `:redraw` :smile:

Reply:
`:redraw` seems to remove the delay

Reply:
However I don't think we should redraw every time. This can cause some visual glitches especially when spamming neogit keybindings. 
![neogit](https://user-images.githubusercontent.com/61095988/113913822-c9c07580-97aa-11eb-843d-67e23ca517ef.gif)

Reply:
Yeah, we definitely need to be smarter about how much data we have to reload